src/aktualizr_info/main.cc

```diff
@@ -91,6 +97,17 @@ int main(int argc, char **argv) {
       }
     }
 
+    std::string director_root;
+    auto registered = false;
+    bool has_metadata = false;
+
+    while (wait_provisioning && (!registered || !has_metadata)) {
+      registered = storage->loadEcuRegistered();
+      has_metadata = storage->loadLatestRoot(&director_root, Uptane::RepositoryType::Director());
+
+      sleep(1);
```

Comment:
This will result in sleeping even if registration and loading the metadata is successful, which is undesirable. I think the previous way you had it where you called those two `load` calls outside of the loop first was actually fine.

Reply:
Ok, I've changed it back and pushed it here. By the way, I didn't change anywhere else but sometimes the CI failed sometimes it passed. It's confusing. I'll check the output again after the CI process is done.

Reply:
About the CI, we've had problem with Travis timing out but it should be much better after some recent changes. If you rebase your PR on master, it should go much faster and fail less often.

Reply:
> About the CI, we've had problem with Travis timing out but it should be much better after some recent changes. If you rebase your PR on master, it should go much faster and fail less often.

It's not the Travis timing out issue. The confusing part is sometimes the prosess will pass.  I'll check the output later.

Reply:
CI is passing this time.

When it failed the message for Travis might be:
The command "if [[ $RELEASE_NAME = ubuntu_16.04 ]]; then ./scripts/test_aktualizr_deb_ubuntu.sh /persistent "${INSTALL_DOCKERFILE}"; fi" exited with 1.
If GitLab CI failed , it's mostly with 'coverage'. 

I don't know if this infomation is useful. I'm sure I'll run into the same situation again. Then I can dig deeper.

Reply:
Sounds like it might be a race condition with the deb generation, probably not related to your PR. If you do see it again, please do let the rest of the team know, though.